[Bug]: terraform-provider-aws_v5.72.1_x5 plugin crashed for resource group

### Terraform Core Version

v1.7.5

### AWS Provider Version

v5.72.1

### Affected Resource(s)

aws_resource_grp

### Expected Behavior

Applying changes should have tagged lambda function to particular resource groups

### Actual Behavior

╷
│ Error: Plugin did not respond
│ 
│   with module.apps.module.order_transform_carrefour_fr_sqs.aws_resourcegroups_resource.lambda[0],
│   on modules/sfn_lambda/main.tf line 49, in resource "aws_resourcegroups_resource" "lambda":
│   49: resource "aws_resourcegroups_resource" "lambda" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-aws_v5.72.1_x5 plugin:

panic: runtime error: index out of range [1] with length 1

goroutine 95 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/resourcegroups.resourceResourceCreate({0x1af8b688, 0xc0049565d0}, 0xc00495e000, {0x18b88420, 0xc001657a00})
	github.com/hashicorp/terraform-provider-aws/internal/service/resourcegroups/resource.go:66 +0x785
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func9(0xc00495e000?, {0x18b88420?, 0xc001657a00})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1af8b688?, {0x1af8b688?, 0xc004941a10?}, 0xd?, {0x18b88420?, 0xc001657a00?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:801 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc002ef1180, {0x1af8b688, 0xc004941a10}, 0xc00465d110, 0xc004641e00, {0x18b88420, 0xc001657a00})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa69
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc001686240, {0x1af8b688?, 0xc004941950?}, 0xc0049038b0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc001cbbd50, {0x1af8b688?, 0xc004941680?}, 0xc0049038b0)
	github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc002309ea0, {0x1af8b688?, 0xc004940ea0?}, 0xc002063ea0)
	github.com/hashicorp/terraform-plugin-go@v0.24.0/tfprotov5/tf5server/server.go:865 +0x3bc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x189cdba0, 0xc002309ea0}, {0x1af8b688, 0xc004940ea0}, 0xc004641b80, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.24.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:545 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0013a0200, {0x1af8b688, 0xc004940e10}, {0x1b005b00, 0xc0018d04e0}, 0xc0049459e0, 0xc001a0cb10, 0x2596f538, 0x0)
	google.golang.org/grpc@v1.66.2/server.go:1394 +0xe2b
google.golang.org/grpc.(*Server).handleStream(0xc0013a0200, {0x1b005b00, 0xc0018d04e0}, 0xc0049459e0)
	google.golang.org/grpc@v1.66.2/server.go:1805 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.66.2/server.go:1029 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 55
	google.golang.org/grpc@v1.66.2/server.go:1040 +0x125

Error: The terraform-provider-aws_v5.72.1_x5 plugin crashed!

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

resource "aws_resourcegroups_resource" "lambda" {
  count     = var.partner_slug != "" ? 1 : 0
  group_arn    = "aws_resourcegroups_group.${var.partner_slug}-${var.env_name}-group.arn"
  resource_arn = module.lambda.lambda_function_arn
}

### Steps to Reproduce

- Add this resource group resource under lambda modules - main.tf
- Make sure to add relevant partner_slug or env_name in variables
- Set variables with proper values in partner specific lambdas functions to

### Debug Output

_No response_

### Panic Output

╷
│ Error: Plugin did not respond
│ 
│   with module.apps.module.order_transform_carrefour_fr_sqs.aws_resourcegroups_resource.lambda[0],
│   on modules/sfn_lambda/main.tf line 49, in resource "aws_resourcegroups_resource" "lambda":
│   49: resource "aws_resourcegroups_resource" "lambda" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-aws_v5.72.1_x5 plugin:

panic: runtime error: index out of range [1] with length 1

goroutine 95 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/resourcegroups.resourceResourceCreate({0x1af8b688, 0xc0049565d0}, 0xc00495e000, {0x18b88420, 0xc001657a00})
	github.com/hashicorp/terraform-provider-aws/internal/service/resourcegroups/resource.go:66 +0x785
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func9(0xc00495e000?, {0x18b88420?, 0xc001657a00})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1af8b688?, {0x1af8b688?, 0xc004941a10?}, 0xd?, {0x18b88420?, 0xc001657a00?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:801 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc002ef1180, {0x1af8b688, 0xc004941a10}, 0xc00465d110, 0xc004641e00, {0x18b88420, 0xc001657a00})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa69
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc001686240, {0x1af8b688?, 0xc004941950?}, 0xc0049038b0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc001cbbd50, {0x1af8b688?, 0xc004941680?}, 0xc0049038b0)
	github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc002309ea0, {0x1af8b688?, 0xc004940ea0?}, 0xc002063ea0)
	github.com/hashicorp/terraform-plugin-go@v0.24.0/tfprotov5/tf5server/server.go:865 +0x3bc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x189cdba0, 0xc002309ea0}, {0x1af8b688, 0xc004940ea0}, 0xc004641b80, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.24.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:545 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0013a0200, {0x1af8b688, 0xc004940e10}, {0x1b005b00, 0xc0018d04e0}, 0xc0049459e0, 0xc001a0cb10, 0x2596f538, 0x0)
	google.golang.org/grpc@v1.66.2/server.go:1394 +0xe2b
google.golang.org/grpc.(*Server).handleStream(0xc0013a0200, {0x1b005b00, 0xc0018d04e0}, 0xc0049459e0)
	google.golang.org/grpc@v1.66.2/server.go:1805 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.66.2/server.go:1029 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 55
	google.golang.org/grpc@v1.66.2/server.go:1040 +0x125

Error: The terraform-provider-aws_v5.72.1_x5 plugin crashed!

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

No

--- a/.changelog/40579.txt
+++ b/.changelog/40579.txt
@@ -1,0 +1,6 @@
+```release-note:bug
+resource/aws_resourcegroups_resource: Fix crash when parsing certain ARN formats
+```
+```release-note:enhancement
+resource/aws_resourcegroups_resource: Add import support
+```

--- a/.changelog/40579.txt
+++ b/.changelog/40579.txt
@@ -1,3 +1,6 @@
+```release-note:note
+resource/aws_resourcegroups_resource: The format of the read-only `id` attribute has changed to prevent inconsistent parsing which resulted in provider crashes under certain conditions. The new format is a comma-delimited string combining `group_arn` and `resource_arn` in their entirety. Configuarations relying on the previous format may need to be updated to continue functioning correctly.
+```
 ```release-note:bug
 resource/aws_resourcegroups_resource: Fix crash when parsing certain ARN formats
 ```

--- a/internal/service/resourcegroups/resource.go
+++ b/internal/service/resourcegroups/resource.go
@@ -8,7 +8,6 @@ import (
 	"errors"
 	"fmt"
 	"log"
-	"strings"
 	"time"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
@@ -21,9 +20,14 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/internal/enum"
 	"github.com/hashicorp/terraform-provider-aws/internal/errs"
 	"github.com/hashicorp/terraform-provider-aws/internal/errs/sdkdiag"
+	"github.com/hashicorp/terraform-provider-aws/internal/flex"
 	"github.com/hashicorp/terraform-provider-aws/internal/slices"
 	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
 	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+const (
+	resourceIDPartCount = 2
 )
 
 // @SDKResource("aws_resourcegroups_resource", name="Resource")
@@ -63,7 +67,11 @@ func resourceResourceCreate(ctx context.Context, d *schema.ResourceData, meta in
 
 	groupARN := d.Get("group_arn").(string)
 	resourceARN := d.Get(names.AttrResourceARN).(string)
-	id := strings.Join([]string{strings.Split(strings.ToLower(groupARN), "/")[1], strings.Split(resourceARN, "/")[1]}, "_")
+	id, err := flex.FlattenResourceId([]string{groupARN, resourceARN}, resourceIDPartCount, false)
+	if err != nil {
+		return sdkdiag.AppendFromErr(diags, err)
+	}
+
 	input := &resourcegroups.GroupResourcesInput{
 		Group:        aws.String(groupARN),
 		ResourceArns: []string{resourceARN},

--- a/internal/service/resourcegroups/resource.go
+++ b/internal/service/resourcegroups/resource.go
@@ -46,6 +46,15 @@ func resourceResource() *schema.Resource {
 			Delete: schema.DefaultTimeout(5 * time.Minute),
 		},
 
+		SchemaVersion: 1,
+		StateUpgraders: []schema.StateUpgrader{
+			{
+				Type:    resourceResourceConfigV0().CoreConfigSchema().ImpliedType(),
+				Upgrade: resourceStateUpgradeV0,
+				Version: 0,
+			},
+		},
+
 		Schema: map[string]*schema.Schema{
 			"group_arn": {
 				Type:     schema.TypeString,

--- a/internal/service/resourcegroups/resource.go
+++ b/internal/service/resourcegroups/resource.go
@@ -37,6 +37,10 @@ func resourceResource() *schema.Resource {
 		ReadWithoutTimeout:   resourceResourceRead,
 		DeleteWithoutTimeout: resourceResourceDelete,
 
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
 		Timeouts: &schema.ResourceTimeout{
 			Create: schema.DefaultTimeout(5 * time.Minute),
 			Delete: schema.DefaultTimeout(5 * time.Minute),
@@ -100,8 +104,14 @@ func resourceResourceRead(ctx context.Context, d *schema.ResourceData, meta inte
 	var diags diag.Diagnostics
 	conn := meta.(*conns.AWSClient).ResourceGroupsClient(ctx)
 
-	output, err := findResourceByTwoPartKey(ctx, conn, d.Get("group_arn").(string), d.Get(names.AttrResourceARN).(string))
+	parts, err := flex.ExpandResourceId(d.Id(), resourceIDPartCount, false)
+	if err != nil {
+		return sdkdiag.AppendFromErr(diags, err)
+	}
+	groupARN := parts[0]
+	resourceARN := parts[1]
 
+	output, err := findResourceByTwoPartKey(ctx, conn, groupARN, resourceARN)
 	if !d.IsNewResource() && tfresource.NotFound(err) {
 		log.Printf("[WARN] ResourceGroups Resource (%s) not found, removing from state", d.Id())
 		d.SetId("")
@@ -112,6 +122,7 @@ func resourceResourceRead(ctx context.Context, d *schema.ResourceData, meta inte
 		return sdkdiag.AppendErrorf(diags, "reading Resource Groups Resource (%s): %s", d.Id(), err)
 	}
 
+	d.Set("group_arn", groupARN)
 	d.Set(names.AttrResourceARN, output.Identifier.ResourceArn)
 	d.Set(names.AttrResourceType, output.Identifier.ResourceType)
 

--- a/internal/service/resourcegroups/resource_migrate.go
+++ b/internal/service/resourcegroups/resource_migrate.go
@@ -1,0 +1,54 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package resourcegroups
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/flex"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func resourceResourceConfigV0() *schema.Resource {
+	// Resource with v0 schema (provider v5.81.0 and below)
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"group_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			names.AttrResourceARN: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			names.AttrResourceType: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceStateUpgradeV0(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+	if rawState == nil {
+		rawState = map[string]interface{}{}
+	}
+
+	// Convert id to comma-delimited string combining group_arn and resource_arn
+	parts := []string{
+		rawState["group_arn"].(string),
+		rawState[names.AttrResourceARN].(string),
+	}
+
+	id, err := flex.FlattenResourceId(parts, resourceIDPartCount, false)
+	if err != nil {
+		return rawState, err
+	}
+	rawState[names.AttrID] = id
+
+	return rawState, nil
+}

--- a/internal/service/resourcegroups/resource_test.go
+++ b/internal/service/resourcegroups/resource_test.go
@@ -40,6 +40,11 @@ func TestAccResourceGroupsResource_basic(t *testing.T) {
 					resource.TestCheckResourceAttrSet(resourceName, names.AttrResourceARN),
 				),
 			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
 		},
 	})
 }

--- a/internal/service/resourcegroups/resource_test.go
+++ b/internal/service/resourcegroups/resource_test.go
@@ -38,8 +38,8 @@ func TestAccResourceGroupsResource_basic(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckResourceExists(ctx, resourceName, &r),
 					resource.TestCheckResourceAttr(resourceName, names.AttrResourceType, "AWS::EC2::Host"),
-					resource.TestCheckResourceAttrPair(resourceName, "group_arn", groupResourceName, "arn"),
-					resource.TestCheckResourceAttrPair(resourceName, names.AttrResourceARN, hostResourceName, "arn"),
+					resource.TestCheckResourceAttrPair(resourceName, "group_arn", groupResourceName, names.AttrARN),
+					resource.TestCheckResourceAttrPair(resourceName, names.AttrResourceARN, hostResourceName, names.AttrARN),
 					resource.TestCheckResourceAttrSet(resourceName, names.AttrID),
 				),
 			},

--- a/internal/service/resourcegroups/resource_test.go
+++ b/internal/service/resourcegroups/resource_test.go
@@ -24,6 +24,8 @@ func TestAccResourceGroupsResource_basic(t *testing.T) {
 	var r types.ListGroupResourcesItem
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_resourcegroups_resource.test"
+	groupResourceName := "aws_resourcegroups_group.test"
+	hostResourceName := "aws_ec2_host.test"
 
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:                 func() { acctest.PreCheck(ctx, t) },
@@ -36,14 +38,53 @@ func TestAccResourceGroupsResource_basic(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckResourceExists(ctx, resourceName, &r),
 					resource.TestCheckResourceAttr(resourceName, names.AttrResourceType, "AWS::EC2::Host"),
-					resource.TestCheckResourceAttrSet(resourceName, "group_arn"),
-					resource.TestCheckResourceAttrSet(resourceName, names.AttrResourceARN),
+					resource.TestCheckResourceAttrPair(resourceName, "group_arn", groupResourceName, "arn"),
+					resource.TestCheckResourceAttrPair(resourceName, names.AttrResourceARN, hostResourceName, "arn"),
+					resource.TestCheckResourceAttrSet(resourceName, names.AttrID),
 				),
 			},
 			{
 				ResourceName:      resourceName,
 				ImportState:       true,
 				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+// Verify the change to the id attribute formatting introduced in v5.82.0
+// do not errors in existing configurations
+func TestAccResourceGroupsResource_v5_82_0_upgrade(t *testing.T) {
+	ctx := acctest.Context(t)
+	var r types.ListGroupResourcesItem
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_resourcegroups_resource.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(ctx, t) },
+		ErrorCheck:   acctest.ErrorCheck(t, names.ResourceGroupsServiceID),
+		CheckDestroy: testAccCheckResourceDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: map[string]resource.ExternalProvider{
+					"aws": {
+						Source:            "hashicorp/aws",
+						VersionConstraint: "5.81.0",
+					},
+				},
+				Config: testAccResourceConfig_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckResourceExists(ctx, resourceName, &r),
+					resource.TestCheckResourceAttrSet(resourceName, names.AttrID),
+				),
+			},
+			{
+				ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+				Config:                   testAccResourceConfig_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckResourceExists(ctx, resourceName, &r),
+					resource.TestCheckResourceAttrSet(resourceName, names.AttrID),
+				),
 			},
 		},
 	})

--- a/website/docs/r/resourcegroups_resource.html.markdown
+++ b/website/docs/r/resourcegroups_resource.html.markdown
@@ -52,3 +52,20 @@ This resource exports the following attributes in addition to the arguments abov
 
 * `create` - (Default `5m`)
 * `delete` - (Default `5m`)
+
+## Import
+
+In Terraform v1.5.0 and later, use an [`import` block](https://developer.hashicorp.com/terraform/language/import) to import an AWS Resource Groups Resource using `group_arn` and `resource_arn`, separated by a comma (`,`). For example:
+
+```terraform
+import {
+  to = aws_resourcegroups_resource.example
+  id = "arn:aws:resource-groups:us-west-2:012345678901:group/example,arn:aws:lambda:us-west-2:012345678901:function:example"
+}
+```
+
+Using `terraform import`, import an AWS Resource Groups Resource using `group_arn` and `resource_arn`, separated by a comma (`,`). For example:
+
+```console
+% terraform import aws_resourcegroups_resource.example arn:aws:resource-groups:us-west-2:012345678901:group/example,arn:aws:lambda:us-west-2:012345678901:function:example
+```

--- a/website/docs/r/resourcegroups_resource.html.markdown
+++ b/website/docs/r/resourcegroups_resource.html.markdown
@@ -30,23 +30,20 @@ resource "aws_resourcegroups_resource" "example" {
   group_arn    = aws_resourcegroups_group.example.arn
   resource_arn = aws_ec2_host.example.arn
 }
-
 ```
 
 ## Argument Reference
 
 The following arguments are required:
 
-* `group_arn` - (Required) The name or the ARN of the resource group to add resources to.
-
-The following arguments are optional:
-
-* `resource_arn` - (Required) The ARN of the resource to be added to the group.
+* `group_arn` - (Required) Name or ARN of the resource group to add resources to.
+* `resource_arn` - (Required) ARN of the resource to be added to the group.
 
 ## Attribute Reference
 
 This resource exports the following attributes in addition to the arguments above:
 
+* `id` - A comma-delimited string combining `group_arn` and `resource_arn`.
 * `resource_type` - The resource type of a resource, such as `AWS::EC2::Instance`.
 
 ## Timeouts

